modal/functions.py

```diff
@@ -954,6 +988,12 @@ async def _load(self: _Function, resolver: Resolver, existing_object_id: Optiona
         obj._is_method = False
         obj._spec = function_spec  # needed for modal shell
 
+        if info.user_cls:
+            obj._method_functions = {}
```

Comment:
I feel like the `_method_functions` should be living on the `_Cls` object instead, since _Function is quite large and complicated as it is, and the methods would only ever get used through the class anyway.